File: src/glove/glove.cpp
#include "glove.h"

Glove::Glove(std::string input_file,
               unsigned long long vocab_size, unsigned long long max_size,
             unsigned long min_count, unsigned long window, unsigned long embed_size, unsigned long epoch,
             int threads, double lr) {
    this->input_file = input_file;
    this->vocab_size = vocab_size;
    this->max_size = max_size;
    this->min_count = min_count;
    this->window = window;
    this->embed_size = embed_size;
    this->epoch = epoch;
    this->threads = threads;
    this->lr = lr;
};

Glove::Glove(std::vector<std::string> args_cmd) {
    parseArgs(args_cmd);
    std::cout << "load cmd ..." << std::endl;
};


Glove::Glove() {

};

void Glove::train(std::string input_file) {
    this->input_file = input_file;

    TRACE("====================================================== ","\n");
    TRACE("building vocabulary","...");
    voca = Vocabulary(min_count, max_size,  input_file);
    vocab_size = voca.size();
    TRACE("vocabulary size is ",vocab_size);
    TRACE("Build vocabulary ","down");



    TRACE("====================================================== ","\n");
    TRACE("building cooccur mat","...");
    CoMat builder(vocab_size,input_file,  window,  memory_limit);
    builder.build(voca);
    TRACE("build cooccur mat  ","down");



    TRACE("====================================================== ","\n");
    model = Train(embed_size, vocab_size, threads,  epoch, lr);
    model.train();
    TRACE("====================================================== ","\n");
};


AnalogyPairs Glove::most_similary(const std::string &word, unsigned long num) {
    return model.most_similary(word, num, voca);
}

void Glove::load( std::string wordvec_file, std::string meta_file) {

    voca.load();

    vocab_size = voca.atoi.size();
    model = Train(embed_size, vocab_size, threads, epoch, lr);
    model.load_model(wordvec_file, meta_file, voca);

}

void Glove::to_txt() {
//    std::cout << "save wordvec..." << std::endl;
    model.to_txt(voca);
//    std::cout << "save wordvec down..." << std::endl;
}

File: python/glove/Glove.py
import os
from glove_pybind import *


class Glove:
    def __init__(self, input_file="small_text",
                vocab_size=0, max_size=10000000, min_count=1,
                 window=10, embed_size=80, epoch=1,  threads=40, lr=0.05):

            self.model = glove()
            self.model.input_file = input_file
            self.model.vocab_size = vocab_size
            self.model.max_size = max_size
            self.model.min_count = min_count
            self.model.window = window
            self.model.epoch = epoch
            self.model.threads = threads
            self.model.lr = lr
            os.mkdir("log/")

    def to_txt(self):
        self.model.to_txt()

    def train(self, file):
        self.model.train(file)

    def most_similary(self, word, num):
        self.model.most_similary(word, num)

    def load(self,  wordvec_file, meta_file):
        self.model.load(wordvec_file, meta_file)




if __name__ == "__main__":
    model = Glove()
    model.train("small_text")
